frontend/src/component/common/Chart/customHighlightPlugin.ts

```diff
@@ -1,8 +1,34 @@
 import type { Chart } from 'chart.js';
 
-export const customHighlightPlugin = (bottomOverflow = 34) => ({
+type CustomHighlightPluginOptions = {
+    width?: number | ((chart: Chart) => number);
+    maxHighlightOpacity?: number;
+    bottomOverflow?: number;
+};
+
+const defaultCategoryPercentage = 0.8;
+export const categoryWidth = (chart: Chart) => {
+    return (
+        (chart.width / chart.scales.x.ticks.length) *
+        (chart.options.datasets?.bar?.categoryPercentage ??
+            defaultCategoryPercentage)
+    );
+};
+
+// Vertical line on the hovered chart, filled with gradient. Highlights a section of a chart when you hover over datapoints
+export const customHighlightPlugin = (
+    options?: CustomHighlightPluginOptions,
+) => ({
     id: 'customLine',
     beforeDraw: (chart: Chart) => {
+        const {
+            width = categoryWidth,
+            maxHighlightOpacity = 0.12,
+            bottomOverflow = 0,
```

Comment:
These seem like sensible defaults to me:
- the highlight should typically be the same width as the category (either a single bar or a group of bars depending on your data source)
- the highlight opacity is the color we'd used everywhere up until now
- no bottom overflow (stop where the chart is).